addons/mrp/models/mrp_production.py

```diff
@@ -1715,15 +1715,30 @@ def _default_amounts(production):
 
             move = moves and moves.pop(0)
             move_qty_to_reserve = move.product_qty
+
+            for index, (quantity, move_line, ml_vals) in enumerate(ml_by_move):
+                taken_qty = min(quantity, move_qty_to_reserve, move_line.qty_done)
```

Comment:
quantity, move_qty_to_reserve is in the product qty but move_line.qty_done in the move line uom